lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'sorbet-runtime'
+require 'concurrent-ruby'
+require 'typhoeus'
+require 'json'
+
+module DevCycle
+  class ConfigManager
+    extend T::Sig
+
+    sig { params(sdkKey: String, local_bucketing: LocalBucketing).returns(NilClass) }
+    def initialize(sdkKey, local_bucketing)
+      @first_load = true
+      @local_bucketing = local_bucketing
+      @sdkKey = sdkKey
+      @config_e_tag = ""
+      @config_poller = Concurrent::TimerTask.new(
+        { execution_interval: @local_bucketing.options.config_polling_interval_ms / 1000, run_now: true }) do |task|
+        fetch_config(false, task)
+      end
+      @config_poller.execute
+      nil
+    end
+
+    def fetch_config(retrying, task)
+      req = Typhoeus::Request.new(
+        get_config_url,
+        headers: {
+          Accept: "application/json",
+        })
+      if @config_e_tag != ""
+        req.headers['If-None-Match'] = @config_e_tag
+      end
+      resp = req.run
+      case resp.code
+      when 304
+        return nil
+      when 200..299
+        @config_e_tag = resp.headers['Etag']
+        return set_config(resp.body)
+      when 403
+        raise("Failed to download DevCycle config; Invalid SDK Key.")
+      when 500...599
+        if !retrying
+          return fetch_config(true, task)
+        end
+        puts("Failed to download DevCycle config. Status: #{resp.code}")
+      else
+        task.shutdown
+        raise("Unexpected response code - DevCycle Response: #{Oj.dump(resp)}")
+      end
```

Comment:
can we cleanup the formatting here? its really hard to read.

```suggestion
      if @config_e_tag != ""
        req.headers['If-None-Match'] = @config_e_tag
      end
      
      resp = req.run
      
      case resp.code
        when 304
          return nil
        when 200..299
          @config_e_tag = resp.headers['Etag']
          return set_config(resp.body)
        when 403
          raise("Failed to download DevCycle config; Invalid SDK Key.")
        when 500...599
          if !retrying
            return fetch_config(true, task)
          end
          puts("Failed to download DevCycle config. Status: #{resp.code}")
        else
          task.shutdown
          raise("Unexpected response code - DevCycle Response: #{Oj.dump(resp)}")
        end
```

Reply:
Not sure why that formatting appears, the indented version is what I see locally. Will poke around.